gateway/src/apicast/policy/upstream/policy.lua

```diff
@@ -0,0 +1,65 @@
+--- Upstream policy
+-- This policy allows to modify the host of a request based on its path.
+
+local resty_resolver = require('resty.resolver')
+local resty_url = require('resty.url')
+local format = string.format
+local ipairs = ipairs
+local tonumber = tonumber
+local match = ngx.re.match
+local table = table
+
+local _M = require('apicast.policy').new('Upstream policy')
+
+local new = _M.new
+
+local function change_upstream(url)
+  ngx.ctx.upstream = resty_resolver:instance():get_servers(
+    url.host, { port = tonumber(url.port) })
```

Comment:
I need to check if `tonumber()` here is needed or if the `resty_url.parse()` call guarantees that we'll get a number here.

Reply:
It does : https://github.com/3scale/lua-resty-url/blob/6fe0896c6f3688d6042a0f28d32ba577cb250515/src/resty/url.lua#L63

Changed.